test/actions.test.js

```diff
@@ -5,350 +5,86 @@ var test = require('tape');
     assert.ok(err instanceof actions.CloudFormationError, 'expected error returned');
     AWS.CloudFormation.restore();
     assert.end();
   });
 });
 
-test('[actions.diff] invalid parameters', function(assert) {
+test('[actions.diff] template url does not exist', function(assert) {
   AWS.stub('CloudFormation', 'createChangeSet', function(params, callback) {
-    var err = new Error('Parameters: [Pets, Age, Name, LuckyNumbers, SecretPassword] must have values');
+    var err = new Error('Template file referenced by https://my-bucket.s3.amazonaws.com/my-template.json does not exist.');
     err.code = 'ValidationError';
     callback(err);
   });
 
-  actions.diff('my-stack', 'us-east-1', 'https://my-bucket.s3.amazonaws.com/my-template.json', {}, function(err) {
+  actions.diff('my-stack', 'us-east-1', 'UPDATE', 'https://my-bucket.s3.amazonaws.com/my-template.json', {}, function(err) {
     assert.ok(err instanceof actions.CloudFormationError, 'expected error returned');
     AWS.CloudFormation.restore();
     assert.end();
   });
 });
 
-test('[actions.diff] template url does not exist', function(assert) {
+test('[actions.diff] template url is invalid', function(assert) {
   AWS.stub('CloudFormation', 'createChangeSet', function(params, callback) {
-    var err = new Error('Template file referenced by https://my-bucket.s3.amazonaws.com/my-template.json does not exist.');
+    var err = new Error('The specified url must be an Amazon S3 URL.');
     err.code = 'ValidationError';
     callback(err);
   });
 
-  actions.diff('my-stack', 'us-east-1', 'https://my-bucket.s3.amazonaws.com/my-template.json', {}, function(err) {
+  actions.diff('my-stack', 'us-east-1', 'UPDATE', 'https://my-bucket.s3.amazonaws.com/my-template.json', {}, function(err) {
     assert.ok(err instanceof actions.CloudFormationError, 'expected error returned');
     AWS.CloudFormation.restore();
     assert.end();
   });
 });
 
-test('[actions.diff] template url is invalid', function(assert) {
+test('[actions.diff] template is invalid', function(assert) {
   AWS.stub('CloudFormation', 'createChangeSet', function(params, callback) {
-    var err = new Error('The specified url must be an Amazon S3 URL.');
+    var err = new Error('Template format error: At least one Resources member must be defined.');
     err.code = 'ValidationError';
     callback(err);
   });
 
-  actions.diff('my-stack', 'us-east-1', 'https://my-bucket.s3.amazonaws.com/my-template.json', {}, function(err) {
+  actions.diff('my-stack', 'us-east-1', 'UPDATE', 'https://my-bucket.s3.amazonaws.com/my-template.json', {}, function(err) {
     assert.ok(err instanceof actions.CloudFormationError, 'expected error returned');
     AWS.CloudFormation.restore();
     assert.end();
   });
 });
 
-test('[actions.diff] template is invalid', function(assert) {
+test('[actions.diff] unexpected createChangeSet error', function(assert) {
```

Comment:
How does this test differ from the next one? Both are labeled as `unexpected createChangeSet error`

Reply:
This test is basically documentation to show how Cloudformation will react on an invalid changeSetType. I'll update the test title to `'[actions.diff] createChangeSet error on wrong changeSetType'`